test/browser/fragments.test.js

```diff
@@ -495,6 +495,7 @@ describe('Fragment', () => {
 		expect(scratch.innerHTML).to.equal('<div>Hello</div>');
 	});
 
+	// TODO: Investigate
```

Comment:
This is the only test in core that is currently failing. Previously we converted array returns into unkeyed Fragments, so an unkeyed Fragment render return and an array return were viewed as the same:
```js
render() { return [<div />]; }
// becomes
c._prevVNode = <Fragment><div /></Fragment>

render() { return <Fragment><div /></Fragment>; }
// becomes
c._prevVNode = <Fragment><div /></Fragment>
```

However, this PR changes that behavior such that if render returns an array we flatten it, and if render returns a Fragment we just put it into the `_children` array :/
```js
render() { return [<div />]; }
// becomes
vnode._children = [<div />];

render() { return <><div /></>; }
// becomes
vnode._children = [<Fragment><div /></Fragment>];
```

Any thoughts on the importance of this feature?

Reply:
In other words, `render()` used to always return a single VNode, and we coerced array returns into Fragments. Now, `render()` always returns an array and no coercion takes place.

Reply:
Here is the relevant check and comment in React for this behavior: https://git.io/fj0P9

Perhaps it wouldn't be too big to add a similar check...

Reply:
Its 21 B to coerce top level non-keyed fragments to arrays.
```js
tmp = c.render(c.props, c.state, c.context);
let isTopLevelFragment = tmp != null && tmp.type == Fragment && tmp.key == null;
toChildArray(isTopLevelFragment ? tmp.props.children : tmp, newVNode._children=[], coerceToVNode, true);
```

Reply:
I went ahead and added this code. Figured it was useful for components like a loader HOC:

```js
render() {
  if (this.state.loading) {
    return <Fragment>Loading...</Fragment> // or something more complex
  }
  else {
    return props.children; // or some other array of children
  }
}
```